Web/stockdogweb/src/components/Listing/Listing.js

```diff
@@ -13,55 +13,65 @@ class Listing extends Component {
 
    generateListings = () => {
       var listingElements = [];
+
+      // Dont' do anything if there are no listings
+      if (!this.props.listings) {
+         return;
+      }
+
       this.props.listings.forEach(listing => {
 
          // Need to do the priceChange element seperately because
          // it changes depending on if its negative or positive
          var priceChangeElement;
 
          if (listing.priceChange >= 0) {
-            priceChangeElement = <div class="listing-item-price-change listing-up-color">
-               (+{listing.priceChange})
-            </div>
+            priceChangeElement = 
+               <div className="listing-item-price-change listing-up-color">
+                  (+{listing.priceChange})
+               </div>
          }
          else {
-            priceChangeElement = <div class="listing-item-price-change listing-down-color">
-               ({listing.priceChange})
-            </div>
+            priceChangeElement = 
+               <div className="listing-item-price-change listing-down-color">
+                  ({listing.priceChange})
+               </div>
          }
 
          // Need to do amount separately since 0 shows nothing
          var amountElement = <div></div>;
```

Comment:
This could just be null if it's an empty div. Don't know if that makes a difference for anything though but it might be cleaner for debugging or something when looking at the dom

Reply:
You're right 👍